v2/common/src/main/java/com/google/cloud/teleport/v2/transforms/ElasticsearchTransforms.java

```diff
@@ -0,0 +1,326 @@
+                      .setFunctionName(options().getTypeFnName())
+                      .build()));
+    }
+
+    /** Builder for {@link WriteToElasticsearch}. */
+    @AutoValue.Builder
+    public abstract static class Builder {
+      public abstract Builder setOptions(WriteToElasticsearchOptions options);
+
+      abstract WriteToElasticsearch autoBuild();
+
+      public WriteToElasticsearch build() {
+
+        WriteToElasticsearch write = autoBuild();
+
+        checkArgument(
+            write.options().getNodeAddresses() != null,
+                "Node address(es) must not be null.");
+
+        checkArgument(write.options().getDocumentType() != null,
+                "Document type must not be null.");
+
+        checkArgument(write.options().getIndex() != null,
+                "Index must not be null.");
+
+        checkArgument(
+            write.options().getBatchSize() > 0,
+            "Batch size must be > 0. Got: " + write.options().getBatchSize());
+
+        checkArgument(
+            write.options().getBatchSizeBytes() > 0,
+            "Batch size bytes must be > 0. Got: " + write.options().getBatchSizeBytes());
+
+        /* Check that both {@link RetryConfiguration} parameters are supplied. */
+        if (write.options().getMaxRetryAttempts() != null
+            || write.options().getMaxRetryDuration() != null) {
+          checkArgument(
+              write.options().getMaxRetryDuration() != null
+                  && write.options().getMaxRetryAttempts() != null,
+              "Both max retry duration and max attempts must be supplied.");
+        }
+
+        if (write.options().getIdFnName() != null || write.options().getIdFnPath() != null) {
+          checkArgument(
+              write.options().getIdFnName() != null && write.options().getIdFnPath() != null,
+              "Both IdFn name and path must be supplied.");
+        }
+
+        if (write.options().getIndexFnName() != null || write.options().getIndexFnPath() != null) {
+          checkArgument(
+              write.options().getIndexFnName() != null && write.options().getIndexFnPath() != null,
+              "Both IndexFn name and path must be supplied.");
+        }
+
+        if (write.options().getTypeFnName() != null || write.options().getTypeFnPath() != null) {
+          checkArgument(
+              write.options().getTypeFnName() != null && write.options().getTypeFnPath() != null,
+              "Both TypeFn name and path must be supplied.");
+        }
+
+        return write;
+      }
+    }
+  }
+
+  /**
+   * Class for routing functions that implements {@link FieldValueExtractFn}. {@link
+   * ValueExtractorFn#apply(JsonNode)} will return null if {@link ValueExtractorFn#functionName()}
+   * or {@link ValueExtractorFn#fileSystemPath()} are null meaning no function is applied to the
+   * document.
+   */
+  @AutoValue
+  public abstract static class ValueExtractorFn implements FieldValueExtractFn {
```

Comment:
This extractor fn is very useful and generic enough to but put inside of its on transform instead of being embedded inside ElasticsearchTransforms. Thoughts?

Reply:
Has been put into a separate transform called ValueExtractorTransform in the common module.